build_util/merge_update_infos.py

```diff
@@ -23,8 +22,8 @@ def merge_json_string(src: str, dst: str) -> str:
     >>> merge_json_string(src, dst)
     '[{"version": "1"}]'
     """
-    src_json: List[Dict[str, Union[str, List[str]]]] = json.loads(src)
-    dst_json: List[Dict[str, Union[str, List[str]]]] = json.loads(dst)
+    src_json: list[dict[str, str | list[str]]] = json.loads(src)
+    dst_json: list[dict[str, str | list[str]]] = json.loads(dst)
```

Comment:
･･･ここ、`str | list[str]`を受け入れ可能にしてるの変な気がしますね･･･。
（あとの処理では`list[str]`しか考えられてない気がする）

とりあえずここにバリデーション入れるFIXMEと、`str | list[str]`で合ってるのか検討するTODO足させてください 🙇 
```suggestion
    # FIXME: バリデーションする
    # TODO: `str | list[str]`だけど`str`が来るとエラーになるのでならないようにしたい
    src_json: list[dict[str, str | list[str]]] = json.loads(src)
    dst_json: list[dict[str, str | list[str]]] = json.loads(dst)
```